server/schedulers/hot_region.go

```diff
@@ -402,52 +402,28 @@ func (bs *balanceSolver) init() {
 		Count: maxCur.Count * bs.sche.conf.GetCountRankStepRatio(),
 	}
 
+	bs.firstPriority, bs.secondPriority = prioritiesToDim(bs.getPriorities())
+}
+
+func (bs *balanceSolver) isSelectedDim(dim int) bool {
+	return dim == bs.firstPriority || dim == bs.secondPriority
+}
+
+func (bs *balanceSolver) getPriorities() []string {
 	// For read, transfer-leader and move-peer have the same priority config
 	// For write, they are different
 	switch bs.rwTy {
 	case read:
-		bs.firstPriority, bs.secondPriority = bs.adjustConfig(bs.sche.conf.GetReadPriorities(), getReadPriorities)
+		return adjustConfig(bs.cluster, bs.sche.conf.GetReadPriorities(), getReadPriorities)
 	case write:
 		switch bs.opTy {
 		case transferLeader:
-			bs.firstPriority, bs.secondPriority = bs.adjustConfig(bs.sche.conf.GetWriteLeaderPriorities(), getWriteLeaderPriorities)
+			return adjustConfig(bs.cluster, bs.sche.conf.GetWriteLeaderPriorities(), getWriteLeaderPriorities)
 		case movePeer:
-			bs.firstPriority, bs.secondPriority = bs.adjustConfig(bs.sche.conf.GetWritePeerPriorities(), getWritePeerPriorities)
+			return adjustConfig(bs.cluster, bs.sche.conf.GetWritePeerPriorities(), getWritePeerPriorities)
 		}
 	}
-}
-
-func (bs *balanceSolver) isSelectedDim(dim int) bool {
-	return dim == bs.firstPriority || dim == bs.secondPriority
-}
-
-// adjustConfig will adjust config for cluster with low version tikv
-// because tikv below 5.2.0 does not report query information, we will use byte and key as the scheduling dimensions
-func (bs *balanceSolver) adjustConfig(origins []string, getPriorities func(*prioritiesConfig) []string) (first, second int) {
-	querySupport := bs.cluster.IsFeatureSupported(versioninfo.HotScheduleWithQuery)
-	withQuery := slice.AnyOf(origins, func(i int) bool {
-		return origins[i] == QueryPriority
-	})
-	compatibles := getPriorities(&compatibleConfig)
-	if !querySupport && withQuery {
-		schedulerCounter.WithLabelValues(bs.sche.GetName(), "use-compatible-config").Inc()
-		return prioritiesToDim(compatibles)
-	}
-
-	defaults := getPriorities(&defaultConfig)
-	isLegal := slice.AllOf(origins, func(i int) bool {
-		return origins[i] == BytePriority || origins[i] == KeyPriority || origins[i] == QueryPriority
-	})
-	if len(defaults) == len(origins) && isLegal && origins[0] != origins[1] {
-		return prioritiesToDim(origins)
-	}
-
-	if !querySupport {
-		schedulerCounter.WithLabelValues(bs.sche.GetName(), "use-compatible-config").Inc()
-		return prioritiesToDim(compatibles)
-	}
-	schedulerCounter.WithLabelValues(bs.sche.GetName(), "use-default-config").Inc()
-	return prioritiesToDim(defaults)
+	return []string{}
```

Comment:
It looks this situation will never happen, do we need a log or some other way to handle it?

Reply:
how about warnning log?

Reply:
Better use an error log?